src/mtcp/Makefile.in

```diff
@@ -82,28 +84,52 @@ endif
 HEADERS = mtcp_header.h mtcp_restart.h mtcp_sys.h mtcp_util.h \
 	  $(srcdir)/../membarrier.h $(DMTCP_INCLUDE_PATH)/procmapsarea.h
 
-OBJS = mtcp_restart.o stdlibfnc.o mtcp_util.o mtcp_check_vdso.o ${ARM_BINARIES}
+OBJS_ORIG = mtcp_restart.o stdlibfnc.o mtcp_util.o mtcp_check_vdso.o \
+            ${ARM_BINARIES}
+OBJS = ${OBJS_ORIG}
 
 ifneq ($(MANA_HELPER_DIR),)
   HEADERS += $(MANA_HELPER_DIR)/mtcp_split_process.h \
              $(MANA_HELPER_DIR)/ucontext_i.h
   OBJS += mtcp_restart_plugin.o mtcp_split_process.o getcontext.o
-  CFLAGS += -DMTCP_PLUGIN_H="<mtcp_restart_plugin.h>"
+  DEFINE_PLUGIN = -DMTCP_PLUGIN_H="<mtcp_restart_plugin.h>"
   INCLUDES += -I$(MANA_HELPER_DIR)
 endif
 
 all: default
 default: build
 libs: build
-build: $(targetdir)/bin/$(MTCP_RESTART)
+build: $(targetdir)/bin/$(MTCP_RESTART) $(targetdir)/bin/$(MTCP_RESTART)_orig \
+                                        $(targetdir)/bin/$(MTCP_RESTART)_MANA
```

Comment:
@gc00  let's make the suffix MANA a variable to make it easy to change the suffix multiple places in the future.

Reply:
In the absence of MANA, does this still try to build mtcp_restart_mana? Shouldn't mtcp_restart_mana only be generated if mana_helper_dir exists?